src/main.rs

```diff
@@ -336,15 +393,25 @@ impl LavaUploadableFile {
             store,
         }
     }
+
+    pub fn artifact(id: i64, path: String, store: Arc<AvailableArtifactStore>) -> Self {
+        Self {
+            which: LavaUploadableFileType::Artifact { id, path },
+            store,
+        }
+    }
 }
 
 impl UploadableFile for LavaUploadableFile {
     type Data<'a> = Box<dyn AsyncRead + Send + Unpin + 'a>;
 
     fn get_path(&self) -> Cow<'_, str> {
-        match self.which {
+        match &self.which {
             LavaUploadableFileType::Log { id } => format!("{}_log.yaml", id).into(),
             LavaUploadableFileType::Junit { id } => format!("{}_junit.xml", id).into(),
+            LavaUploadableFileType::Artifact { id, path } => {
+                format!("{}_artifacts/{}", id, path).into()
```

Comment:
I'd just make that path as it can only come out of `submit` jobs; the <job_id>_bla.* mostly came due to those coming from the monitor jobs as well